test/support/utils_sycl.h

```diff
@@ -397,6 +392,152 @@ test4buffers(int mult = kDefaultMultValue)
 {
     test4buffers<alloc_type, typename TestName::UsedValueType, TestName>(mult);
 }
+
+
+// Device copyable noop functor used in testing as surrogate for predicates, binary ops, unary functors
```

Comment:
Do we want to keep it here? Or do we want to create some `test/device_copyable_utils.h`? I don't have strong preference to either and I don't want to hold this PR any longer but I am a little bit worried about manageability of our files. Think about this comment as of asking your opinion. I won't push for any of those.

Reply:
These helpers are very specific and not useful for other purposes, so perhaps it is better to create a new file..  To keep with our naming conventions, perhaps `test/utils_device_copyable.h` would be better.

Reply:
I've separated them out.